api/wasm/v0/cpp/proxy_wasm_externs.h

```diff
@@ -0,0 +1,121 @@
+ * one or more related plugins and their corresponding proxy_on_configure(), proxy_on_done() and
+ * proxy_on_delete() calls. It is also be provided during creation of stream context for those
+ * plugins.
+ * @param configuration_size is the size of any configuration available via proxy_get_configuration.
+ * @return non-zero on success and zero on failure (e.g. bad configuration).
+ */
+extern "C" uint32_t proxy_on_start(uint32_t root_context_id, uint32_t configuration_size);
+/**
+ * Can be called to validate a configuration (e.g. by an xDS provider) both before proxy_on_start()
+ * to verify the VM configuration or after proxy_on_start() to verify a plugin configuation.
+ * @param root_context_id is a unique identifier for the configuration verification context.
+ * @param configuration_size is the size of any configuration available via
+ * proxy_get_configuration().
+ * @return non-zero on success and zero on failure (i.e. bad configuration).
+ */
+extern "C" uint32_t proxy_validate_configuration(uint32_t root_context_id,
+                                                 uint32_t configuration_size);
+/**
+ * Called when a plugin loads or when plugin configuration changes dynamically.
+ * @param root_context_id is an identifier for one or more related plugins.
+ * @param configuration_size is the size of any configuration available via
+ * proxy_get_configuration().
+ * @return non-zero on success and zero on failure (e.g. bad configuration).
+ */
+extern "C" uint32_t proxy_on_configure(uint32_t root_context_id, uint32_t configuration_size);
+
+// Stream calls.
+
+/**
+ * Called when a request, stream or other ephemeral context is created.
+ * @param context_id is an identifier the ephemeral context.
+ * @param configuration_size is the size of any configuration available via
+ * proxy_get_configuration().
+ */
+extern "C" void proxy_on_create(uint32_t context_id, uint32_t root_context_id);
+
+// Stream and Non-stream calls.
+
+/**
+ * For stream contexts, called when the stream has completed. Note: if applicable proxy_on_log() is
+ * called after proxy_on_done() and before proxy_on_delete(). For root contexts, proxy_on_done() is
+ * called when the VM is going to shutdown.
+ * @param context_id is an identifier the context.
+ * @return non-zero to indicate that this context is done. Stream contexts must return non-zero.
+ * Root contexts may return zero to defer the VM shutdown and the proxy_on_delete call until after a
+ * future proxy_done() call by the root context.
+ */
+extern "C" uint32_t proxy_on_done(uint32_t context_id);
+/**
+ * Called when the context is being deleted and will no longer receive any more calls.
+ * @param context_id is an identifier the context.
+ */
+extern "C" void proxy_on_delete(uint32_t context_id);
+
+//
+// ABI calls from the VM to the host.
+//
+
+// Configuration and Status
+
+/**
+ * Called from the VM to get any configuration. Valid only when in a proxy_on_start(),
```

Comment:
So, this is VM-level configuration vs. plugin configuration? Can you add references to where in the API this lives?

Reply:
Updated comments.  PTAL